### PR TITLE
Bluespace Miner Config

### PR DESCRIPTION
**Pull request by francinum**
LRP can have their crutch if they really want it.

This at the very least removes the problem from MRP, who should have more than enough cooperation and players to not require the help.

## About The Pull Request
Creates the config entries:

`roundstart_bluespace_miners` , amount of boards in the RD's locker at roundstart
`bsminer_researchable`, if the currently existing node is available for research.

## Changelog
:cl:
config: Bluespace miner is now affected by config. Golden has two miners and cannot research more. Sage has no miners and cannot research more.
/:cl:

closes #3390

### CODE REVIEW

**Comment by Cenrus** [comment]
Nice

**Comment by ghost** [comment]
Half measures are always worse, just saying.

**Comment by francinum** [comment]
> Half measures are always worse, just saying.

I'm aware. This is merely an alternative provided.

I'm still for putting a bullet in the feature whole hog.

**Comment by ike709** [comment]
Wouldn't it be better to just restrict it to lowpop? Assuming this is the kind of approach we want to take.

**Comment by jlsnow301** [comment]
> Wouldn't it be better to just restrict it to lowpop? Assuming this is the kind of approach we want to take.

Tie the output to the server pop?

**Comment by WindowsErrors** [comment]
we did it lads, we saved beestation

**Comment by Archanial** [comment]
If I had to choose between it being config value or completely removed I'd say removal is better.
If someone has better rework idea they can just revert the pr.

**Comment by PerishedFraud** [comment]
good. +1

**Comment by FortSumter** [comment]
Better idea: keep on both servers, but add a check to see how many players on connected. If the hard cap (lets say 30 players connected) is reached, then the bluespace miner research will not be available. A system like this already exists for certain jobs roundstart access (IE the bartender having access to botany and the kitchen at roundstart if no one signed up for those jobs), and so it shouldn't be so hard to implement. While you're at it, you can add back the autodoc and connect it to the same check, that way we have the ability to implant ourselves again if there is no one able to do surgery (seeing how dogshit self surgery is currently, having a massive failure rate even when using proper tools).

**Comment by francinum** [comment]
> Better idea: keep on both servers, but add a check to see how many players on connected. If the hard cap (lets say 30 players connected) is reached, then the bluespace miner research will not be available. A system like this already exists for certain jobs roundstart access (IE the bartender having access to botany and the kitchen at roundstart if no one signed up for those jobs), and so it shouldn't be so hard to implement. While you're at it, you can add back the autodoc and connect it to the same check, that way we have the ability to implant ourselves again if there is no one able to do surgery (seeing how dogshit self surgery is currently, having a massive failure rate even when using proper tools).

Cool. Get me code for this in a way that can't be gamed or abused.

**Review by PowerfulBacon — APPROVED** [review]

**Review by PowerfulBacon — DISMISSED** [review]
Maybe the boards should come in a box to not spam the locker

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -606,13 +606,23 @@
...
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
 	export_price = 5000
 
+/datum/techweb_node/bluespace_mining/New() //Techweb nodes don't have an init,
+	. = ..()
+
+	if(CONFIG_GET(flag/bsminer_researchable))
+		hidden = FALSE
+	else
+		hidden = TRUE
```
Comment (park66665) [inline code/modules/research/techweb/all_nodes.dm]:
```suggestion
	hidden = !CONFIG_GET(flag/bsminer_researchable)
```

**Review by Crossedfall — DISMISSED** [review]
Merging this with the option to nerf and adjust the config options later

**Review by Crossedfall — APPROVED** [review]

**Pull request merged**